src/main/kotlin/uk/gov/justice/digital/hmpps/gqlapi/config/AuthTokenFilterFunction.kt

```diff
@@ -0,0 +1,31 @@
+package uk.gov.justice.digital.hmpps.gqlapi.config
+
+import org.springframework.http.HttpHeaders
+import org.springframework.web.reactive.function.client.ClientRequest
+import org.springframework.web.reactive.function.client.ClientResponse
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction
+import org.springframework.web.reactive.function.client.ExchangeFunction
+import reactor.core.CoreSubscriber
+import reactor.core.publisher.Mono
+
+class AuthTokenFilterFunction : ExchangeFilterFunction {
+
+  override fun filter(request: ClientRequest, next: ExchangeFunction): Mono<ClientResponse> {
+
+    return AuthTokenResponseMono(request, next)
+  }
+}
+
+class AuthTokenResponseMono(
+  private val request: ClientRequest,
+  private val next: ExchangeFunction
+) : Mono<ClientResponse>() {
+
+  override fun subscribe(subscriber: CoreSubscriber<in ClientResponse>) {
```

Comment:
flux equivalent to filter, except it looks in reactor context rather than thread local